btree/btree.c

```diff
@@ -2942,14 +3003,15 @@ static int64_t btree_put_tick(struct m0_sm_op *smop)
 				else
 					bop->bo_flags |= BOF_LOCKALL;
 			}
-			if (bop->bo_arbor->t_height <= tree->t_height) {
+			if (bop->bo_arbor->t_height < tree->t_height) {
```

Comment:
If the height changes (increases or decreases) you can free all the allocations and start over. The assumption is that the height change is a rare occurrence.